src/sql-parser/src/parser.rs

```diff
@@ -338,7 +338,29 @@ impl<'a> Parser<'a> {
             }
             Token::Keyword(id) => self.parse_qualified_identifier(id.into_ident()),
             Token::Ident(id) => self.parse_qualified_identifier(Ident::new(id)),
-            Token::Op(op) if op == "+" || op == "-" => Ok(Expr::Op {
+            Token::Op(op) if op == "-" => {
+                if let Some(Token::Number(n)) = self.peek_token() {
+                    let n = match n.parse::<f64>() {
```

Comment:
That's just for my understanding: we're trying to parse as `f64` because all number literals we support can be parsed as that?

Reply:
Yep, that was the idea. Because I'm just checking for it being equal to `0`, I am being a little cavalier with the check.